### PR TITLE
Loop

### PR DESCRIPTION
**Pull request by tonyblackbird**
Did todays lecture assignments

### CODE REVIEW

**Comment by mlastovski** [comment]
Gut

File: lecture_notes/07 basics/0-buggy.cpp [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
+#include <iostream>
 
 int main() {
   int firstNum = 2;
   int secondNum = 3;
   int sum = firstNum + secondNum;
-  printf("%d\n", sum);
+  int a = 0;
+  std::cin >> a;
+  std::cout << a+2;
```
Comment (tshemsedinov) [inline lecture_notes/07 basics/0-buggy.cpp:9]: Just a code style `a + 2`

File: lecture_notes/07 basics/4-loop.cpp [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#include <stdio.h>
+#include <string>
+
+int sum(int argument[], int length) {
+  int output = 0;
+  for(int i = 0; i < length; i++){
```
Comment (tshemsedinov) [inline lecture_notes/07 basics/4-loop.cpp:6]: Consistent style: `for (int i = 0; i < length; i++) {`

File: lecture_notes/07 basics/4-loop.cpp [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    output += argument[i];
+  }
+  return output;
+}
+
+int main() {
+  int test_input[] = {1, 3, 5, 7, 9};
+  int length = sizeof(test_input) / sizeof(int);
+  std::string output = std::to_string(sum(test_input, length));
```
Comment (tshemsedinov) [inline lecture_notes/07 basics/4-loop.cpp:15]: For such a long constructions just for readability use intermediate variables:
```
int total = sum(test_input, length);
std::string output = std::to_string(total);
```

File: lecture_notes/07 basics/4-loop.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+function sum(...args){
+    let output = 0;
+    for(const number of args){
```
Comment (tshemsedinov) [inline lecture_notes/07 basics/4-loop.js:3]: ditto

File: lecture_notes/07 basics/4-loop.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+function sum(...args){
```
Comment (tshemsedinov) [inline lecture_notes/07 basics/4-loop.js:1]: Add space `) {`

Comment (tshemsedinov) [reply lecture_notes/07 basics/4-loop.js:1]: And I'd propose to use arrow functions where possible

**Review by tshemsedinov — COMMENTED** [review]
First prize 🥇

**Pull request merged**